dist/ng-obiba-mica.js

```diff
@@ -1114,7 +1112,292 @@ angular.module('obiba.mica.access')
 
       return this;
     }]);
-;angular.module('templates-ngObibaMica', ['access/views/data-access-request-form.html', 'access/views/data-access-request-histroy-view.html', 'access/views/data-access-request-list.html', 'access/views/data-access-request-validation-modal.html', 'access/views/data-access-request-view.html', 'attachment/attachment-input-template.html', 'attachment/attachment-list-template.html']);
+;/*
+ * Copyright (c) 2014 OBiBa. All rights reserved.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the GNU Public License v3.0.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+'use strict';
+
+function GraphicChartsDataProvider() {
+
+  function DataProvider(dataResponse) {
+    var data = dataResponse;
+    this.getData = function (callback) {
+      data.$promise.then(callback);
+    };
+  }
+
+  this.$get = function (GraphicChartsGetData, GraphicChartsConfig) {
+    return new DataProvider(GraphicChartsGetData.get({id: GraphicChartsConfig.getOptions().IdNetworks}));
+  };
+}
+
+angular.module('obiba.mica.graphics', [
+    'googlechart',
+    'obiba.utils',
+    'templates-ngObibaMica'
+  ])
+  .config(['$provide', function ($provide) {
+    $provide.provider('GraphicChartsData', GraphicChartsDataProvider);
+  }]);
+;/*
+ * Copyright (c) 2014 OBiBa. All rights reserved.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the GNU Public License v3.0.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+'use strict';
+
+angular.module('obiba.mica.graphics')
+
+  .directive('graphicsGeoCharts', [function () {
+    return {
+      restrict: 'EA',
+      replace: true,
+      scope: {
+        fieldTransformer: '@',
+        chartType: '@',
+        chartAggregationName: '@',
+        chartEntityDto: '@',
+        chartOptionsName: '@'
+      },
+      templateUrl: 'graphics/views/charts-directive.html',
+      controller: 'GraphicChartsController'
+    };
+  }])
+
+  .directive('graphicsSelectionCriteria', [function () {
```

Comment:
All these directives should have generic names and by type, i.e., obibaGeoChart, obibaPieChart, obibaBarChart, etc